assets/src/edit-story/app/userOnboarding/useUserOnboarding.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+/**
+ * Internal dependencies
+ */
+// import { useCurrentUser } from '../currentUser';
+// import { useHelpCenter } from '../helpCenter';
+
+const TRIGGER_CONTEXTS = {
+  SAFE_ZONE: 'safeZone',
+  // todo [@embarks]: other onboarding triggers
+};
+
+export function useUserOnboarding(selector) {
+  const triggerType = selector(TRIGGER_CONTEXTS);
+
+  // const { updateCurrentUser, currentUser } = useCurrentUser(
+  //   ({ state, actions }) => ({
+  //     currentUser: state.currentUser,
+  //     updateCurrentUser: actions.updateCurrentUser,
+  //   })
+  // );
+
+  // const { goToTip } = useHelpCenter(({ actions }) => ({
+  //   goToTip: actions.goToTip,
+  // }));
```

Comment:
I think we'll need to create a dedicated action in the help center that's like `openUpToTipIfUnread(TIP_KEY)` (pardon my inability to come up with a good name lol 😅).

Only reason I say that is I think the more apparent things it will do are opening up the modal and setting the current tip index on the same render as the modal opening (so we don't see it transition to the tip as it's opening), but also to set some sort of flag so that a lot of the inference and reseting effects are opted out of on that specific action.

basically have it return some state partial like this
```js
(tipKey) => isTipKeyUnread(tipKey) && ({
  isOpen: true,
  navigationIndex: getIndexFromKey(tipKey),
  // some flag like this to opt out of any resetting or smart behavior we have in the effects
  isOpeningToIndex: true
})
```